src/main/java/net/discordjug/javabot/listener/filter/MessageFilterHandler.java

```diff
@@ -0,0 +1,110 @@
+package net.discordjug.javabot.listener.filter;
+
+import lombok.RequiredArgsConstructor;
+import net.discordjug.javabot.data.config.BotConfig;
+import net.discordjug.javabot.systems.moderation.AutoMod;
+import net.discordjug.javabot.util.ExceptionLogger;
+import net.discordjug.javabot.util.WebhookUtil;
+import net.dv8tion.jda.api.entities.channel.ChannelType;
+import net.dv8tion.jda.api.entities.channel.attribute.IWebhookContainer;
+import net.dv8tion.jda.api.entities.channel.middleman.StandardGuildChannel;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * Message filtering handler.
+ */
+@RequiredArgsConstructor
+public class MessageFilterHandler extends ListenerAdapter {
+
+	private final List<MessageFilter> filters;
+	private final AutoMod autoMod;
+	private final BotConfig botConfig;
+
+	@Override
+	public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+		if (!shouldRunFilters(event)) {
+			return;
+		}
+
+		MessageContent content = new MessageContent(event,
```

Comment:
`Message#getAttachments` and `Message#getEmbeds` return unmodifiable `List`s (not allowing modifications). Please change it to `new ArrayList<>(event.getMessage().getAttachments())` and `new ArrayList<>(event.getMessage().getEmbeds())`.